lightblue/src/main/java/org/esbtools/eventhandler/lightblue/LightblueDocumentEventRepository.java

```diff
@@ -103,7 +103,28 @@ public void addNewDocumentEvents(Collection<? extends DocumentEvent> documentEve
                 .map(LightblueDocumentEventRepository::asEntity)
                 .collect(Collectors.toList());
 
-        lightblue.data(InsertRequests.documentEventsReturningOnlyIds(documentEventEntities));
+        int newEventsCount = documentEventEntities.size();
+        int maxEventsPerInsert = config.getOptionalMaxDocumentEventsPerInsert().orElse(newEventsCount);
+
+        int numberOfRequestsOfMaxEvents = newEventsCount / maxEventsPerInsert;
+        int remainder = newEventsCount % maxEventsPerInsert;
+
+        int requests = remainder > 0
```

Comment:
If i follow your math right, I think 109 - 114 can be simplified to something like Math.ceiling(newEventsCount / maxEventsPerInsert) maybe with some type casting thrown in.